SharpMap/Rendering/Decoration/ScaleBar/ScaleBarAltUnit.cs

```diff
@@ -0,0 +1,127 @@
+﻿using System;
+
+namespace SharpMap.Rendering.Decoration.ScaleBar
+{
+    /// <summary>
+    /// Scale Bar map decoration with switching between units for large and small scales
+    /// </summary>
+    [Serializable]
+    public class ScaleBarAltUnit : ScaleBar
+    {
+        //NB defaults set during base class initialisation
+        private int _barUnitAlt1;
+        private int _barUnitAlt2;
+
+        private double _barUnitFactorAlt1;
+        private double _barUnitFactorAlt2;
+
+        private Boolean _forceIt = false;
+
+        /// <summary>
+        /// ScaleBar unit. Changing BarUnit will reset BarUnitAlt1 and BarUnitAlt2 accordingly.
+        /// </summary>
+        public override int BarUnit
+        {
+            get => base.BarUnit;
+            set
+            {
+                base.BarUnit = value;
+                // This effectively reverts to standard ScaleBar behaviour.
+                _barUnitAlt1 = BarUnit;
+                _barUnitFactorAlt1 = _barUnitFactor;
+                _barUnitAlt2 = BarUnit;
+                _barUnitFactorAlt2 = _barUnitFactor;
+            }
+        }
+
+        /// <summary>
+        /// Bar Unit for use at large scales (small area) such as ft, m, or yd
+        /// </summary>
+        public int BarUnitAlt1
```

Comment:
I must admit, I'm not comfortable with `BarUnit1` and `BarUnit2` as property names.
Longer but much more descriptive would be `BarUnitLargeScale` and `BarUnitSmallScale`.

Reply:
No problems re property names. However, I'm new to C# and thought I was familiar with extension methods but now I'm a bit confused and wondering if I have understood your comment. 

The way it is at the moment, **ScaleBarAltUnit** overrides `CalcBarScale` that is called during `OnRender`, and I thought an extension method cannot override class methods. I've done a bit more digging around to see if I'm missing something, but haven't been able to find anything just yet. Please could you give me some links so I can take another look, or a brief outline of how you would like it to be?

Reply:
I didn't mean extension methods, I mean make the changes required to have `ScaleBar` behave just like your `ScaleBarAlt` class.